client/reader/discover/controller.js

```diff
@@ -73,8 +90,27 @@ const exported = {
 		/* eslint-enable wpcalypso/jsx-classname-namespace */
 		next();
 	},
+
+	discoverSsr( context, next ) {
+		const state = context.store.getState();
+		if ( ! isUserLoggedIn( state ) ) {
+			context.renderHeaderSection = renderHeaderSection;
+		}
```

Comment:
SSR won't run for logged-in users, so this check should be unnecessary:

```suggestion
		context.renderHeaderSection = renderHeaderSection;
```

Reply:
Good point, fixed!